scripts/setup_env.py

```diff
@@ -7,10 +7,11 @@
 
 target_mip = param["OUTPUT_MIP"]
 image_resolution = param["IMAGE_RESOLUTION"]
-if image_resolution[0] == image_resolution[1] == image_resolution[2]:
-    mip0_factor = [2**target_mip] * 3
-else:
-    mip0_factor = [2**target_mip, 2**target_mip, 1]
+
+param["ISOTROPIC"] = param.get("ISOTROPIC", False)
+isotropic = param["ISOTROPIC"] or (image_resolution[0] == image_resolution[1] == image_resolution[2])
```

Comment:
I think you do not need to modify `param["ISOTROPIC"]` permanently here, just use get in line 12

Reply:
Actually I think we should allow `param["ISOTROPIC"]` to override the heuristic unconditionally, I mean treating the data as anisotropic if `param["ISOTROPIC"]` is explicitly false, even if the input resolution happen to be the same

Reply:
That sounds fair.